### PR TITLE
fix(terraform_plan): fix module resources ids

### PR DESCRIPTION
**Pull request by YaaraVerner**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description
use the address as the resource id if the `CHECKOV_ENABLE_NESTED_MODULES` env variable is enabled.
add the last referred module (if it exists) to the resource id if the `CHECKOV_ENABLE_NESTED_MODULES` env variable is off.
use the old plan resource id for plan enrichment from terraform dir (without deep analysis)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/output/report.py [hunk 1]
```diff
@@ -593,6 +595,13 @@ def handle_skipped_checks(
                 report.failed_checks.remove(record)
         return report
 
+    @staticmethod
+    def get_plan_resource_raw_id(resource_id: str) -> str:
+        resource_raw_id = ".".join(resource_id.split(".")[-2:])
```
Comment (bo156) [inline checkov/common/output/report.py:600]: didn't you did something just like that somewhere else?

File: checkov/terraform/plan_utils.py [hunk 2]
```diff
@@ -102,3 +102,7 @@ def get_entity_context(definitions, definitions_raw, definition_path, full_file_
                 entity_context['address'] = resource_defintion[TF_PLAN_RESOURCE_ADDRESS]
                 return entity_context
     return entity_context
+
+
+def get_resource_id_without_nested_modules(address: str) -> str:
+    return ".".join(address.split(".")[-4:])
```
Comment (bo156) [inline checkov/terraform/plan_utils.py:108]: the random numbers are a bit wierd for anyone not familiar with the build, can you use a const?

File: checkov/common/output/report.py [hunk 3]
```diff
@@ -593,6 +595,13 @@ def handle_skipped_checks(
...
+        if '[' in resource_raw_id:
```
Comment (achiar99) [inline checkov/common/output/report.py:601]: i think it will be better to wrap with the env var

**Review by bo156 — APPROVED** [review]

**Review by achiar99 — APPROVED** [review]

File: checkov/common/output/report.py (hunk 1 above)
Comment (gruebel) [reply checkov/common/output/report.py:600]: a doc string with an example would be great 🙂

Comment (YaaraVerner) [reply checkov/common/output/report.py:600]: no

File: checkov/terraform/plan_utils.py (hunk 2 above)
Comment (gruebel) [reply checkov/terraform/plan_utils.py:108]: also here a doc string with an example, then it more clear, what actually is happening

File: checkov/terraform/runner.py [hunk 4]
```diff
@@ -241,6 +242,11 @@ def get_graph_checks_report(self, root_folder: str, runner_filter: RunnerFilter,
                             if referrer_id:
                                 resource = f'{referrer_id}.{resource_id}'
                         definition_context_file_path = get_tf_definition_key_from_module_dependency(full_file_path, module_dependency, module_dependency_num)
+                    elif entity.get(CustomAttributes.TF_RESOURCE_ADDRESS) and entity.get(CustomAttributes.TF_RESOURCE_ADDRESS) != resource_id:
+                        # for plan resources
+                        resource = entity.get(CustomAttributes.TF_RESOURCE_ADDRESS)
```
Comment (gruebel) [inline checkov/terraform/runner.py:247]:
```suggestion
                        resource = entity[CustomAttributes.TF_RESOURCE_ADDRESS]
```

**Review by gruebel — COMMENTED** [review]
nice work 💪

**Review by gruebel — APPROVED** [review]

**Pull request merged**